### PR TITLE
Document, Index, and Prettify Graphics, Part 1 (A-H)

### PR DESCRIPTION
**Pull request by hjk321**
The major goal of this pull request is to get the vast majority, if not all, of FireRed's graphics in a fully editable state. This means:

- If the gfx are documented in the codebase but not in the folder, I will document it. If the gfx is documented in the folder and not the codebase, I will document it. Otherwise, I will leave it undocumented unless it is obvious at first glance what it is.
- Any graphics that are still in .bin format (excluding tilemaps, obviously) will be converted to 4bpp or equivalent.
- Tilesets that are in ridiculous dimensions (eg. 8 x 672) will be resized to something more manageable, usually 8 tiles wide, but depending on the situation may be 16 or 4 wide.
- Any images still in grayscale format will be indexed with their corresponding palette (or at least one of them) for easier editing.
- If a palette is exclusive to a certain image, I will remove pal files in favor of generating the palette from the image itself.

I will be going through each subfolder in the graphics directory in alphabetical order. Do not expect this pull request to be finished anytime soon.

Because of the sheer amount of graphics that require processing, I will only be going for the "low-hanging fruit" for now. Essentially, any image that can be processed with gbagfx's command line will be done, but for special cases that require extra time or additional research to properly document or index, I will skip them and list it here to be tackled at a later time.

### Skip List:
- Some healthbox and status stuff in `battle_interface`. Apparently the palettes are dynamic which makes them a headache to index properly.
- Credits gfx, because I believe it's already being worked on by someone else right now.
- Some door animations that I couldn't find the palette for. Also I was lazy.
- Documentation for some field effects and all field specials, because I don't know most of them.

### CODE REVIEW

**Comment by hjk321** [comment]
Note: The previous commit moves the `graphics/battle` folder to `graphics/battle_terrain` and the `graphics/battle_terrain` to `graphics/battle_terrain/unused`. This is because the `battle` folder contained the actual, used graphics for battle terrain, while `battle_terrain` contained unused files (the majority of which were not even included in the rom).

**Comment by hjk321** [comment]
Ready to review.

**Comment by GriffinRichards** [comment]
This makes the `graphics/battle_terrain` folder much worse. For one they were previously conveniently grouped into sub-folders, now they're all in one folder and prefixed unnecessarily. They're also visually worse now. For example, the cave terrain before:
Screen Shot 2021-04-12 at 12 43 46 PM
and after:
Screen Shot 2021-04-12 at 12 44 12 PM

File: src/battle_interface.c [hunk 1]
```diff
@@ -342,7 +342,7 @@ static const struct SubspriteTable gUnknown_8260404[] = {
     {NELEMS(gUnknown_82603E4), gUnknown_82603E4}
 };
 
-static const u16 gUnknown_26040C[] = INCBIN_U16("graphics/battle_interface/unk_826404C.4bpp");
+static const u16 gBattleInterface_Unused[] = INCBIN_U16("graphics/battle_interface/unused.4bpp");
```
Comment (GriffinRichards) [inline src/battle_interface.c:345]: Prefix `s` for static symbols

File: src/diploma.c [hunk 2]
```diff
@@ -29,9 +29,9 @@ static void Task_WaitForExit(u8);
 static void Task_DiplomaInit(u8);
 static void Task_DiplomaReturnToOverworld(u8);
 
-static const u32 gUnknown_84147C0[] = INCBIN_U32("graphics/diploma/unk_84147C0.4bpp.lz");
-static const u32 gUnknown_84154E8[] = INCBIN_U32("graphics/diploma/unk_84154E8.bin.lz");
-static const u16 gUnknown_8415954[] = INCBIN_U16("graphics/diploma/unk_8415954.gbapal");
+static const u32 gDiplomaGfx[] = INCBIN_U32("graphics/diploma/diploma.4bpp.lz");
+static const u32 gDiplomaTilemap[] = INCBIN_U32("graphics/diploma/diploma.bin.lz");
+static const u16 gDiplomaPal[] = INCBIN_U16("graphics/diploma/diploma.gbapal");
```
Comment (GriffinRichards) [inline src/diploma.c:34]: Same, prefix `s`

**Comment by hjk321** [comment]
Ready to merge.

**Pull request merged**